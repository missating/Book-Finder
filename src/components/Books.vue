<template>
  <v-container>
    <v-container grid-list-xl>
      <v-layout wrap>
        <v-flex xs6 v-for="(book, index) in books" :key="index">
          <book-card v-if="book" :book="book"></book-card>
        </v-flex>
      </v-layout>
    </v-container>
  </v-container>
</template>

<script>
import BookCard from "@/components/BookCard.vue";

export default {
  name: "Books",
  components: {
    BookCard
  },
  data() {
    return {};
  },
  computed: {
    books() {
      return this.$store.getters.getBooks;
    },
  }
};
</script>
