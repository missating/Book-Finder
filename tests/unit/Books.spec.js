import Vuex from 'vuex';
import Vue from 'vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Books from '@/components/Books';
import Vuetify from 'vuetify';

Vue.use(Vuetify);
const localVue = createLocalVue();
localVue.use(Vuex);

describe('Books.vue', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        books: [],
        searchQuery: '',
        isFetching: false,
        apiBaseUrl: 'https://www.googleapis.com/books/v1/volumes',
      },
      getters: {
        books: state => state.books,
      },
    });
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Books, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });
});
