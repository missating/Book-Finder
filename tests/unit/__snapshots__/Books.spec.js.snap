// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books.vue should render correctly 1`] = `
<v-container-stub
  tag="div"
>
  <v-container-stub
    grid-list-xl=""
    tag="div"
  >
    <v-layout-stub
      tag="div"
      wrap=""
    />
  </v-container-stub>
</v-container-stub>
`;
